app/domain/authentication/authn_k8s/set_file_content_in_container.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    SetFileContentInContainer ||= CommandClass.new(
+      dependencies: {
+        kubectl_exec:      KubectlExec.new,
+        k8s_object_lookup: K8sObjectLookup,
+        logger:            Rails.logger
+      },
+      inputs: %i(webservice pod_namespace pod_name path content mode container)
+    ) do
+
+      LOG_FILE = "${TMPDIR:-/tmp}/conjur_set_file_content.log"
+
+      def call
+        set_file_content_in_container
+      end
+
+      private
+
+      def set_file_content_in_container
+        @kubectl_exec.call(
+          k8s_object_lookup: @k8s_object_lookup.new(@webservice),
+          pod_namespace:     @pod_namespace,
+          pod_name:          @pod_name,
+          container:         @container,
+          cmds:              %w(sh),
+          body:              set_file_content_script(@path, @content, @mode),
+          stdin:             true
+        )
+      end
+
+      # Sets the content of a file in a given path to the given content
+      # We first copy the content into a temporary file and only then move it to
+      # the desired path as the client polls on its existence and we want it to
+      # exist only when the whole content is present.
+      #
+      # We redirect the output to a log file on the authn-client container
+      # that will be written in its logs for supportability.
+      def set_file_content_script(path, content, mode)
+        tmp_cert = "#{path}.tmp"
+
+        "
+#!/bin/sh
+set -e
+
+cleanup() {
+ rm -rf \"#{tmp_cert}\"
```

Comment:
Do we have some chance to raise condition here so we delete the directory during the set_file_content?

Reply:
delete which directory? `/etc/conjur/ssl`? if so - why would we delete it?

Reply:
I meant if there is a possibility that we will run set_file_content and on the same time we will run cleanup (different thread)  (i know its a matter of msec..)

Reply:
i don't follow. if we write content to a file, and also delete it in parallel - then how can the content exist on the file?

Furthermore - the temp file is also in the `ssl` directory which is limited to the running user

Reply:
Its clear to me know .. rm -rf confused me (-r is used for directory, why you need this? ) its not critical anyway.

Reply:
thanks, i removed the `-r` flag